examples/ai-core/src/e2e/feature-test-suite.ts

```diff
@@ -676,6 +677,188 @@ export function createFeatureTestSuite({
                   expect((await result.usage).totalTokens).toBeGreaterThan(0);
                 }
               });
+
+              it('should handle multiple sequential tool calls', async () => {
+                let calculatorCalls = 0;
+                const result = await generateText({
+                  model,
+                  prompt:
+                    'First calculate 2+2, then calculate the result times 3 using the calculator tool.',
+                  tools: {
+                    calculator: {
+                      parameters: z.object({
+                        expression: z.string(),
+                      }),
+                      execute: async ({ expression }) => {
+                        calculatorCalls++;
+                        return String(eval(expression));
+                      },
+                    },
+                  },
+                  maxSteps: 10,
+                });
+
+                expect(result.text).toContain('12'); // (2+2)*3 = 12
+                expect(calculatorCalls).toBe(2); // Should be called exactly twice
+              });
+
+              it('should handle tool execution errors gracefully', async () => {
+                try {
+                  const result = await generateText({
+                    model,
+                    prompt: 'Calculate 1/0 using the calculator tool.',
+                    tools: {
+                      calculator: {
+                        parameters: z.object({
+                          expression: z.string(),
+                        }),
+                        execute: async ({ expression }) => {
+                          const result = eval(expression);
+                          if (!isFinite(result))
+                            throw new Error('Invalid calculation');
+                          return String(result);
+                        },
+                      },
+                    },
+                  });
+
+                  // Allow "gracefully" to mean either (a) the model refuses to
+                  // execute the tool call, or (b) the tool call fails.
```

Comment:
The model only generates the tool call (which will happen), the rest is done by AI SDK core.